backend/api/views.py

```diff
@@ -130,99 +145,136 @@ def put(self, request, cashflow_id):
-                'description': 'Deletes user that corresponds to the Access Token send in the header'
-            },
-            {
-                'Endpoint': '/users/get',
-                'method': 'GET',
-                self.tokenNeeded: 'Access',
-                'body': None,
-                'description': 'Returns user that corresponds to the Access Token send in the header'
-            },
-            {
-                'Endpoint': '/users/update',
-                'method': 'PUT',
-                self.tokenNeeded: 'Access',
-                'body': {'email': '', 'newPassword': '', 'oldPassword': ''},
-                'description': 'Not implemented yet. Updates current user with data sent in put request'
-            },
-        ]
-        return Response(routes)
+
+    def post(self, request):
+        try:
+            account = Account.objects.get(id=request.data['account'])
+            private = Private.objects.create(account=account, first_name=request.data['private']['first_name'],
+                                             last_name=request.data['private']['last_name'])
+        except Exception as e:
+            # Fixme differentiate between other exception types
+            print(e.__cause__)
+            return JsonResponse({'success': False}, status=status.HTTP_400_BAD_REQUEST)
+
+        return JsonResponse({'success': True, 'private': private.id}, status=status.HTTP_201_CREATED)
+
+    def get(self, request, private_id):
+        try:
+            private = Private.objects.get(id=private_id)
+        except Exception as e:
+            # Fixme differentiate between other exception types
+            print(e.__cause__)
+            return JsonResponse({'success': False}, status=status.HTTP_404_NOT_FOUND)
+
+        serializer = PrivateSerializer(private, many=False)
+        return Response(serializer.data, status=status.HTTP_200_OK)
+
+
+class ItemView(APIView):
+    # permission_classes = (IsAuthenticated,)
+
+    def post(self, request, cashflow_id):
+        try:
+            cashflow = Cashflow.objects.get(id=cashflow_id)
+            item = Item.objects.create(name=request.data['name'], amount=request.data['amount'],
+                                       value=request.data['value'],
+                                       cashflow=cashflow)
+        except Exception as e:
+            # Fixme differentiate between other exception types
+            print(e.__cause__)
+            return JsonResponse({'success': False}, status=status.HTTP_400_BAD_REQUEST)
+
+        return JsonResponse({'success': True, 'item': item.id}, status=status.HTTP_201_CREATED)
+
+    def get(self, request, cashflow_id, item_id):
+        try:
+            item = Item.objects.get(id=item_id)
+            if item.cashflow.id != cashflow_id:
+                raise AttributeError('invalid item and cashflow combination')
+        except Exception as e:
+            # FIXME: Differentiate between ObjectDoesNotExist and a broad Exception
```

Comment:
should this be tackled in this PR?

Reply:
No, the FIXME should not be tackled in this PR.

In case if you meant the following lines:

```python
 if item.cashflow.id != cashflow_id:
                raise AttributeError('invalid item and cashflow combination')
```

I think it is not wrong to add this exception there as I had to create this whole ItemView class in order to support items and it is also already tested in test_views. Are you worried about reducing the size of this PR or an out of scope action?